### DISCUSSION TITLE
init_iface is not been called

### DISCUSSION THREAD

**Discussion by ofirshe** [zephyrproject-rtos/zephyr#53342] [Q&A] [answered]
Hey, 

I developed an ethernet driver for our custom board and I included ETH_NET_DEVICE_DT_INST_DEFINE macro at the end of the file:

ETH_NET_DEVICE_DT_INST_DEFINE(n,				\
				kinetis_ethernet_xsight_eth_init,					\
			    NULL,			\
			    &eth##n##_context,				\
			    &eth##n##_buffer_config,			\
			    CONFIG_ETH_INIT_PRIORITY,			\
			    &api_funcs,					\
			    NET_ETH_MTU);

I see that the entry function kinetis_ethernet_xsight_eth_init I passed in ETH_NET_DEVICE_DT_INST_DEFINE macro s being called in debugging.

That's my api_funcs struct: 
static const struct ethernet_api api_funcs = {
	.iface_api.init		= kinetis_ethernet_xsight_eth_iface_init,
	.get_capabilities	= kinetis_ethernet_xsight_get_capabilities,
	.set_config		= kinetis_ethernet_xsight_set_config,
	.send			= kinetis_ethernet_xsight_eth_tx,
};

I see that zephyr doesn't call api_funcs.iface_api.init (kinetis_ethernet_xsight_eth_iface_init) function and the context's iface remains empty for my driver.

Does someone maybe know what can cause this problem ?

Thanks,
Ofir.

**Comment by ofirshe** [comment, marked as answer]
closed

**Reply by iLyngklip** [reply]
Did you figure out what caused this?

BR Mathias
